### PR TITLE
Remove unnecessary protocol checks

### LINKED ISSUE #2818: Remove Unnecessary Protocol 15 Checks
When implementing protocol 15, we added several protocol version checks to preserve the behavior of protocol version 14 as a safety precaution. With the protocol 15 upgrade behind us, these checks have no value and should be removed.

### PR DESCRIPTION
**Pull request by jonjove**
# Description

Resolves #2818

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]
just one question in there

File: src/transactions/ManageOfferOpFrameBase.cpp [hunk 1]
```diff
@@ -590,7 +590,7 @@ ManageOfferOpFrameBase::doCheckValid(uint32_t ledgerVersion)
         // Note: This was not invalid before version 3
     }
 
-    if (ledgerVersion >= 15 && mOfferID < 0)
+    if (mOfferID < 0)
```
[the remark this answers is not in the document]
Comment (jonjove) [reply src/transactions/ManageOfferOpFrameBase.cpp:593]: I wasn't 100% sure when I wrote this, but I recall that we wrote that with the belief that a negative offer ID had never actually come up. But I investigated, and this is not true. There is a [single operation](https://horizon.stellar.org/operations/125697062640680961) that was submitted with a negative offer ID. I'll revert this change.

File: src/ledger/LedgerTxn.cpp [hunk 2]
```diff
@@ -3223,13 +3223,6 @@ LedgerTxnRoot::Impl::getNewestVersion(InternalLedgerKey const& gkey) const
     }
     catch (NonSociRelatedException& e)
     {
-        if (getHeader().ledgerVersion <= 14)
```
[the remark this answers is not in the document]
Comment (jonjove) [reply src/ledger/LedgerTxn.cpp:3226]: Fixed.

**Comment by MonsieurNicolas** [comment]
r+ a7aac63910102584eb3fb87713aa545ee9807646

**Pull request merged**